pynest/nest/lib/hl_api_connection_helpers.py

```diff
@@ -236,3 +248,61 @@ def fixdict(d):
     projections = fixdict(projections)
     sps(projections)
     sr('ConnectLayers')
+
+
+def _check_input_nodes(pre, post, conn_spec):
+    '''
+    Check the properties of `pre` and `post` nodes:
+
+    * if `conn_spec` is "one_to_one", skip uniqueness check for better speed
+      and use data_connect
```

Comment:
This is only true if pre or post is already an array, right?

Reply:
no, if one or both are ``NodeCollection`` objects, the whole test is skipped, so of course the uniqueness is also skipped (since we know the uniqueness has to be fulfilled by such an object)

Reply:
Yes, but if both are NodeCollections it can never use `connect_arrays`, as both `pre_is_nc` and `post_is_nc` will be `True`. This bullet point makes it seem like if `conn_spec` is "one_to_one", `connect_arrays` will always be used. It's the `and use data_connect` that I find confusing.

Reply:
To me the important bit is to let people reading the docstring know when a unicity test will be performed and when not (for performance reason).
If I understand correctly, your issue is actually with

> and use data_connect

so should I just remove that part and put something like

> if `conn_spec` is "one_to_one", no uniqueness check is performed and regular one_to_one or data_connect is used depending on the input type